exercises/subprograms/challenge_119.py

```diff
@@ -0,0 +1,68 @@
+"""
+Define a subprogram that will ask the user to pick a low and a high number,
+and then generate a random number between those two values and store it in
+a variable called  “comp_num”.
+
+Define another  subprogram that will  give the instruction “I am  thinking
+of a number…” and then ask the user to guess the number they are thinking of.
+
+Define a third subprogram that will check to see if the comp_num is the same
+as the user’s guess. If it is, it should display the message “Correct, you
+win”, otherwise it should keep looping, telling the user if they are too low or
+too high and asking them to guess again until they guess correctly.
+"""
+
+import random
+import common_functions
+
+
+def ask_for_user_numbers():
+    while True:
+        num_low = common_functions.get_user_input_integer(
+            "Please, enter the minimum number: "
+        )
+        num_high = common_functions.get_user_input_integer(
+            "Please, enter the maximum number: "
+        )
+
+        if num_low < num_high:
+            comp_num = random.randrange(num_low, num_high)
+            return comp_num
+
+        print(
```

Comment:
Can you connect this error to a validation? Example, if the code gets changed and another error situation was introduced, it would still print this message.

Reply:
I agree. I think probably inverting the logic will simplify this as the happy path can return a valid number

Reply:
I have inverted the logic, and indeed the code is more simple. Thanks for the suggestion.
I didn't get the point about connecting the error to validation. Should I use exceptions?

Reply:
No, you already did it with the flow change. By using a formula like `if error_condition: show_error_message()` it's obvious the connection between the error condition and the message. If you do `if not error_condition: return value` and then lower in the code you have `show_error_message()` then you've disconnected the two. See that I mean?